src/lib/util/view.h

```diff
@@ -0,0 +1,154 @@
+// license:BSD-3-Clause
+// copyright-holders:Nathan Woods
+/******************************************************************************
+
+	view.h
+
+	STL container for a view
+
+***************************************************************************/
+
+#pragma once
+
+#ifndef __VIEW_H__
+#define __VIEW_H__
+
+#include <cstddef>
+
+namespace util {
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+template<typename T >
+class view
+{
+public:
+	typedef std::ptrdiff_t difference_type;
+	typedef std::size_t size_type;
+	typedef T value_type;
+	typedef T &reference;
+	typedef const T &const_reference;
+	typedef T *pointer;
+
+	class iterator
+	{
+	public:
+		typedef std::ptrdiff_t difference_type;
+		typedef T value_type;
+		typedef T &reference;
+		typedef T *pointer;
+
+		typedef std::random_access_iterator_tag iterator_category;
+
+		iterator(T *ptr) : m_ptr(ptr) { }
+		iterator(const iterator &that) : m_ptr(that) { }
+
+		iterator& operator=(const iterator &that) { m_ptr = that.m_ptr; return *this; }
+		bool operator==(const iterator &that) const { return m_ptr == that.m_ptr; }
+		bool operator!=(const iterator &that) const { return m_ptr != that.m_ptr; }
+		bool operator<(const iterator &that) const { return m_ptr < that.m_ptr; }
+		bool operator>(const iterator &that) const { return m_ptr > that.m_ptr; }
+		bool operator<=(const iterator &that) const { return m_ptr <= that.m_ptr; }
+		bool operator>=(const iterator &that) const { return m_ptr >= that.m_ptr; }
+		iterator operator+(size_type i) const { return iterator(m_ptr + i); }
+		iterator operator-(size_type i) const { return iterator(m_ptr - i); }
+		difference_type operator-(const iterator &that) const { return m_ptr - that.m_ptr; }
+
+		iterator& operator++() { m_ptr++; return *this; }
+		iterator operator++(int) { iterator result = *this; m_ptr++; return result; }
+
+		reference operator*() const { return *m_ptr; }
+		pointer operator->() const { return m_ptr; }
+		operator T*() const { return m_ptr; }
```

Comment:
The implicit cast operator also appears to be the cause of the "ambiguous conversion" error in the tea-ci build.